app/main.py

```diff
@@ -1,3 +1,21 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Callable:
```

Comment:
The return type of `wrapper` can be any, because you return the result of calculations, not the function itself.